src/renderer/state.ts

```diff
@@ -208,6 +209,19 @@ export class AppState {
     versions: this.baseVersions,
   });
 
+  // Used for communications between windows
+  public broadcastChannel: AppStateBroadcastChannel = new BroadcastChannel(
+    'AppState',
+  );
+
+  // Notifies other windows that this version has changed so they can update their state to reflect that.
+  private updateVersionDownloadStatus(version: RunnableVersion) {
```

Comment:
Let's name this a more generic `updateVersionState`, since it really updates the overall state - at the moment that might be tightly coupled to download status, but any change to the info that `RunnableVersion` contains could make this method name inaccurate.